src/views/List.jsx

```diff
@@ -1,21 +1,29 @@
 import { useEffect, useState } from 'react';
 import { ListItem, SearchBar } from '../components';
 import { useNavigate } from 'react-router-dom';
+import { comparePurchaseUrgency } from '../api/firebase';
+import { getIndicator } from '../utils/helpers';
 
 export function List({ data, listPath }) {
 	const [search, setSearch] = useState('');
+	const [allData, setAllData] = useState([]);
 	const [displayData, setDisplayData] = useState([]);
 	const navigate = useNavigate();
 
 	useEffect(() => {
-		setDisplayData([...data]);
+		const arrayWithIndicator = data.map((item) => ({
+			...item,
+			indicator: getIndicator(item),
+		}));
+		setAllData([...comparePurchaseUrgency(arrayWithIndicator)]);
+		setDisplayData([...comparePurchaseUrgency(arrayWithIndicator)]);
```

Comment:
i think its best to use a variable to be stored in the states instead of calling `comparePurchaseUrgency` twice.